### ISSUE TITLE
some IDs are duped in some cases

### ISSUE THREAD

**Issue by eads** [newsdev/elex#176]
```
2016-01-06 16:11:02,646 (INFO) elex (v0.2.0) : Getting results for election 2015-11-03
ERROR:  duplicate key value violates unique constraint "results_pkey"
DETAIL:  Key (id)=(21281-None-None) already exists.
CONTEXT:  COPY results, line 2784

Fatal error: local() encountered an error (return code 1) while executing 'elex results 2015-11-03 | psql elections16 -c "COPY results FROM stdin DELIMITER ',' CSV HEADER;"'
```

**Comment by eads** [comment]
annnd another:

```
2016-01-06 16:12:17,918 (INFO) elex (v0.2.0) : Getting results for election 2012-11-06
ERROR:  duplicate key value violates unique constraint "results_pkey"
DETAIL:  Key (id)=(0-polid-893-1) already exists.
CONTEXT:  COPY results, line 2975

Fatal error: local() encountered an error (return code 1) while executing 'elex results 2012-11-06 | psql elections16 -c "COPY results FROM stdin DELIMITER ',' CSV HEADER;"'
```

**Comment by jeremyjbowers** [comment]
Ugh.

**Comment by jeremyjbowers** [comment]
Wait. This might be bad data, e.g., initialization data. Might explain why I've never seen it with IA / NH / Super Tuesday.

**Comment by jeremyjbowers** [comment]
The problem for these IDs is that the reporting unit is not available, which is only true for initialization data. And, of course, when I check now, there's the `initialization_data` flag set to True.

**Comment by jeremyjbowers** [comment]
I am going to check this with today's Super Tuesday test.

**Comment by jeremyjbowers** [comment]
Yeah, looking at Super Tuesday and IA results (I have good copies of those; my NH copy is kinda weak), there's proper IDs for everyone.

**Comment by jeremyjbowers** [comment]
You know, I was just looking at https://github.com/nprapps/elections16/blob/master/fabfile/data.py#L56 and I think you want to call `elex %foo% $RACEDATE -t` to make sure you're passing the `test` flag to the AP.

**Comment by jeremyjbowers** [comment]
```select id, first, last, fips, COUNT(*) from elex_results group by id, first, last, fips having COUNT(*) > 1;
found 35 rows```

**Comment by jeremyjbowers** [comment]
That's fun.

**Comment by jeremyjbowers** [comment]
They're all contained to MA and VT. This makes me think something weird is happening with the county/township rollup.

**Comment by jeremyjbowers** [comment]
Here's an example row with 13 duplicates.

<img width="684" alt="screen shot 2016-01-10 at 9 34 55 am" src="https://cloud.githubusercontent.com/assets/109988/12222030/6e6426ea-b77d-11e5-8ba8-d556b48c6136.png">

**Comment by jeremyjbowers** [comment]
Well, found out where it's happening. Everything before https://github.com/newsdev/elex/blob/master/elex/api/api.py#L614 looks right; everything after it looks wrong.

```
{'officename': u'President', 'raceid': u'24547', 'precinctsreporting': 0, 'national': True, 'officeid': u'P', 'id': None, 'fipscode': None, 'reportingunitname': None, 'seatnum': None, 'lastupdated': None, 'initialization_data': False, 'racetype': u'Primary', 'candidates': [<elex.api.api.CandidateReportingUnit object at 0x114271e10>, <elex.api.api.CandidateReportingUnit object at 0x114271e50>, <elex.api.api.CandidateReportingUnit object at 0x114271e90>, <elex.api.api.CandidateReportingUnit object at 0x114271ed0>, <elex.api.api.CandidateReportingUnit object at 0x114271f10>, <elex.api.api.CandidateReportingUnit object at 0x114271f50>, <elex.api.api.CandidateReportingUnit object at 0x114271f90>, <elex.api.api.CandidateReportingUnit object at 0x114271fd0>, <elex.api.api.CandidateReportingUnit object at 0x114278050>, <elex.api.api.CandidateReportingUnit object at 0x114278090>, <elex.api.api.CandidateReportingUnit object at 0x1142780d0>, <elex.api.api.CandidateReportingUnit object at 0x114278110>, <elex.api.api.CandidateReportingUnit object at 0x114278150>, <elex.api.api.CandidateReportingUnit object at 0x114278190>, <elex.api.api.CandidateReportingUnit object at 0x1142781d0>], 'test': True, 'electiondate': None, 'racetypeid': u'R', 'uncontested': False, 'description': None, 'precinctsreportingpct': 0.0, 'reportingunitid': None, 'precinctstotal': 0, 'statepostal': u'MA', 'level': 'county', 'statename': 'Massachusetts', 'votecount': 0, 'seatname': None}
```

**Comment by jeremyjbowers** [comment]
This was really, really nasty. But I got it. Because counties are synthetic, they don't have camelCased keys like the original AP objects would -- and when we switched our `__init__` process, it stopped assigning non-camelCased keys to the proper attributes.

Closed with sha c9d5d18151e324a7ff30aa058ac447aa83addfec. No longer see dupes.

**Comment by jeremyjbowers** [comment]
Nope, found one other case where there is still some duplication. Back on it.

**Comment by jeremyjbowers** [comment]
Okay, caught that case. Now to write a test before I close the ticket. sha 9afe785ee4e283b9f88896b053fef8d1d7b3017b.

**Comment by jeremyjbowers** [comment]
Typoed my `closes` message, but closed for good in sha 818698f93ace9b12b45cbb4d0945f17553d9a4d7.
